src/site_store.rs

```diff
@@ -1,89 +1,91 @@
-use std::ffi::OsString;
 use std::fs::{create_dir_all, read_dir, read_to_string, remove_file, write};
+use std::path::{Path, PathBuf};
 
 use crate::ChangeKind;
 
-#[derive(Clone)]
-pub struct SiteStore {
-    folder: String,
-}
-
-impl SiteStore {
-    pub fn new(folder: String) -> std::io::Result<Self> {
-        create_dir_all(&folder)?;
-        Ok(Self { folder })
-    }
-
-    pub fn remove_gone(&self, expected_basenames: &[String]) -> anyhow::Result<Vec<OsString>> {
+pub fn remove_gone(expected_paths: &[PathBuf]) -> anyhow::Result<Vec<PathBuf>> {
```

Comment:
Could you add a comment what this function does? Alternatively, rename the function to be more descriptive.

Reply:
Something like this?

```suggestion
// Remove site files which are no longer configured to cleanup the directory
pub fn remove_gone(expected_paths: &[PathBuf]) -> anyhow::Result<Vec<PathBuf>> {
```